app/Application/Services/OpenWalletService.php

```diff
@@ -24,6 +24,11 @@ public function createWallet(string $userId): ?string
             return null;
         }
 
-        return $this->walletDataSource->saveWalletInCache();
+        $walletId = $this->walletDataSource->saveWalletInCache();
+        if ($walletId === null) {
```

Comment:
Above you are using is_null. Change this and use the function or replace the above is_null with `$user === null`